js/imgur.js

```diff
@@ -26,8 +26,40 @@ weechat.factory('imgur', ['$rootScope', function($rootScope) {
 
     // Upload image to imgur from base64
     var upload = function( base64img, callback ) {
-        // Set client ID (Glowing Bear)
-        var clientId = "164efef8979cd4b";
+
+        //
+        // Getting the access token
+        //
+        // First, you need a client id, go here to register an app: https://api.imgur.com/oauth2/addclient
+        //
+        // Go to: https://api.imgur.com/oauth2/authorize?client_id=<client id>&response_type=token&state=askin-01
+        //
+        // Accept and it sould redirect you with an url like this: https://<website.tld>/?state=askin-01#access_token=<token here>&expires_in=315360000&token_type=bearer&refresh_token=<refresh token>&account_username=<your nick>&account_id=<your id>
+        //
+        // You can now input this token in the main page to get images uploaded to your account
+        // To get the albumHash, go to imgur.com and create a new album / go to an existing one and pick the string in the url: https://imgur.com/a/<string>
+        //
+
+        var accessToken = "";
+        var albEnabled = "";
+        var albumHash = "";
+        var isClientID = "";
+
+        // Set client ID (Yours)
+        if(localStorage.getItem("iToken").length < 38){
+            accessToken = "164efef8979cd4b"; // Glowing bear default token
+            isClientID = 'true';
```

Comment:
In any case, you should be using booleans (`true` and `false`) instead of strings (`'true'` and `'false'`) for `isClientID`

Reply:
Corrected.